### ISSUE TITLE
Bud new model

### ISSUE THREAD

**Issue by JusticeN** [livebud/bud#238]
Is there a plan to have a command for generating models and connecting it with database?

**Comment by Bishwas-py** [comment]
Yupp, that'll be a lot awesome...

**Comment by matthewmueller** [comment]
Hey folks, thanks for your interest in Bud! We're working on adding built-in model support, but in the meantime, you can do this today yourself.

Conceptually, take a look at the [Dependency Injection docs](https://www.notion.so/Hey-Bud-4d81622cc49942f9917c5033e5205c69#976e4b0666f34d5eae6ccbce93111025) to see how you can bring models into your controllers. Then you can take a look at a real-world example using xo in this https://github.com/vito/bass-loop

**Comment by JusticeN** [comment]
Thank you really good example.

I was thinking about something like the ruby and rails model field generator from the cli.

Or the jhipster one for generating Java spring boot apps. 

Or may be "bud" just have a different philosophy

**Comment by matthewmueller** [comment]
We definitely want to have a Rails-like model generator once we have model support. The philosophy is similar here.

I hadn't heard of jhipster, I'll check that one out, thanks for sharing!

**Comment by Bishwas-py** [comment]
> We definitely want to have a Rails-like model generator once we have model support. The philosophy is similar here.
> 
> I hadn't heard of jhipster, I'll check that one out, thanks for sharing!

We got the model?

**Comment by preslavrachev** [comment]
@matthewmueller I am a fan of SQLBoiler. I suppose it will work just the same, correct? I can simply generate my models and use the dependency injection to pass a `sql.DB` connection to all my controllers.

**Comment by matthewmueller** [comment]
Yep, that would work! You might just need to wrap it to tell Bud how to initialize it. I usually put these wrappers in internal, but it doesn't really matter.

Perhaps in `internal/postgres/postgres.go`:

```go
package postgres

func Open() (*sql.DB, error) {
  // Open handle to database like normal
  return sql.Open("postgres", "dbname=fun user=abc")
}
```

You can also couple it with an environment parser like https://github.com/caarlos0/env to be able to set this config programmatically:

```go
import "app.com/internal/env"

func Open(env *env.Env) (*sql.DB, error) {
  // Open handle to database like normal
  return sql.Open(env.Database.Name, env.Database.URL)
}
```

Then in your `internal/env`:

```go
package env

import "https://github.com/caarlos0/env"

type Env struct {
  Database Database
}

type Database struct {
  Name string `env:"DATABASE_NAME"`
  URL string `env:"DATABASE_URL"`
}

func Parse() (*Env, error) {
  var e Env
  if err :=  env.Parse(&e); err != nil {
    return nil, err
  }
  return &e, nil
}
```

---

While you can do this kind of stuff today, I plan on baking in environment variables after I wrap up this iteration on views.
